pygmt/helpers/utils.py

```diff
@@ -172,8 +172,16 @@ def build_arg_string(kwargs):
     ...     )
     ... )
     -BWSen -Bxaf -Byaf -I1/1p,blue -I2/0.25p,blue -JX4i -R1/2/3/4
+    >>> print(build_arg_string(dict(R="1/2/3/4", J="X4i", watre=True)))
+    Traceback (most recent call last):
+      ...
+    pygmt.exceptions.GMTInvalidInput: Unrecognized parameter 'watre'.
     """
     gmt_args = []
+    # raise an exception for unrecognized options
+    for key in kwargs:
+        if len(key) > 2:
+            raise GMTInvalidInput(f"Unrecognized parameter '{key}'.")
     # Exclude arguments that are None and False
```

Comment:
Perhaps we can rewrite the whole function like below, so we don't need to loop over `kwargs` three times.
```python
    for key in filtered_kwargs:
        # raise an exception for unrecognized options
        if len(key) > 2:
            raise GMTInvalidInput(f"Unrecognized parameter '{key}'.")       
        if (kwargs[key] is None or kwargs[key] is False):
            pass  # Exclude arguments that are None and False
        elif is_nonstr_iter(kwargs[key]):
            for value in kwargs[key]:
                gmt_args.append(f"-{key}{value}")
        elif kwargs[key] is True:
            gmt_args.append(f"-{key}")
        else:
            gmt_args.append(f"-{key}{kwargs[key]}")
    return " ".join(sorted(gmt_args))
```

Reply:
Yep, looks good. It might conflict with #1487 but we can fix it later.